File: 0925-long-pressed-name/0925-long-pressed-name.ts
function isLongPressedName(name: string, typed: string): boolean {
    const name_arr = name.split('')
    const typed_arr = typed.split('')
    const typed_len = typed.length
     if (typed_len < name.length) {
        return false
    }
    for (let i = 0; i < typed_len; i++) {
        if (name_arr[i] === typed_arr[i]) {
        } else {
            if (typed_arr[i] === typed_arr[i - 1]) {
                typed_arr.splice(i, 1)
                i--
            } else {
                return false
            }
        }
    }
    return true
};

File: 0925-long-pressed-name/0925-long-pressed-name.py
class Solution:
    def isLongPressedName(self, name: str, typed: str) -> bool:
        count = 0
        for c in range(len(typed)):
            if count < len(name) and name[count] == typed[c]:
                count += 1
            elif c == 0 or typed[c] != typed[c - 1]:
                return False
        return count == len(name)
